src/url.js

```diff
@@ -71,7 +69,7 @@ export function parseUrl(url) {
 
   const fromCache = cache[url];
   if (fromCache) {
-    return fromCache;
+    return Object.assign({}, fromCache);
```

Comment:
https://github.com/ampproject/amphtml/blob/master/3p/integration.js#L466
Here `parentLocation` comes from `parseUrl` and is cached. 
Do you mean we want to differ from first created url object and cached url object?